kibot/var_kibom.py

```diff
@@ -6,10 +6,12 @@
 """
 Implements the KiBoM variants mechanism.
 """
+from __future__ import annotations
 from .optionable import Optionable
 from .gs import GS
 from .misc import IFILT_MECHANICAL
 from .fil_base import BaseFilter
+from .var_base import BaseVariant
```

Comment:
This isn't needed. The `@variant_class` decorator does it.

Reply:
And the same is for all the other similar imports.

Reply:
And again: adding type hinting in all these places to catch what? A potential problem that never hits the user?

Reply:
May not be needed because of '@variant_class' but is it allowed?

This is apparently not (currently) detected by (all) "linting" tools so they complain and can't detect which attributes/function exist are overloaded, etc (mypy).

So if one wants a tool like mypy to provide usefull hints, it's needed.

Reply:
> So if one wants a tool like mypy to provide usefull hints, it's needed.

Doesn't the tool allow exclusions? Like the `noqa` for flake8, note that macros are there to simplify the code and linting tools can't analyze them.